### PR TITLE
feat: expose registry name mapping methods

### LINKED ISSUE #72: Expose the registry name mapping methods
As users may still need to do the mapping from `docker.io` to `https://index.docker.io/v1/` in their code when using `Store.Get()`, `Store.Put()` and `Store.Delete()`, we can consider expose the existing mapping methods for their convenience.

We may export the following methods and may rename them if needed:
* https://github.com/oras-project/oras-credentials-go/blob/97227b115704e3a102099b92910541e0e1e95cda/registry.go#L82-L90
* https://github.com/oras-project/oras-credentials-go/blob/97227b115704e3a102099b92910541e0e1e95cda/registry.go#L92-L100

### PR DESCRIPTION
**Pull request by wangxiaoxuan273**
Resolves #72

### CODE REVIEW

**Review by TerryHowe — APPROVED** [review]
/lgtm

File: registry.go [hunk 1]
```diff
@@ -71,15 +71,15 @@ func Logout(ctx context.Context, store Store, registryName string) error {
...
+		reg = MapRegistryNameRegistry1DockerIo(reg)
 		if reg == "" {
 			return auth.EmptyCredential, nil
 		}
 		return store.Get(ctx, reg)
 	}
 }
 
-func mapStoreRegistryName(registry string) string {
+func MapRegistryNameDockerIo(registry string) string {
```
Comment (shizhMSFT) [inline registry.go:82]: The method name is not meaningful.

Comment (shizhMSFT) [reply registry.go:82]: This method maps a registry name to a key for credential store. Therefore, `func ServerAddressFromRegistry(registry string) string` with documentation is preferable.

File: registry.go [hunk 2]
```diff
@@ -89,7 +89,7 @@ func mapStoreRegistryName(registry string) string {
 	return registry
 }
 
-func mapAuthenticationRegistryName(hostname string) string {
+func MapRegistryNameRegistry1DockerIo(hostname string) string {
```
Comment (shizhMSFT) [inline registry.go:92]: The method name is not meaningful.

Comment (shizhMSFT) [reply registry.go:92]: This method maps a host name to a key for credential store. Therefore, `func ServerAddressFromHostname(hostname string) string` with documentation is preferable.

File: registry.go (hunk 1 above)
Comment (wangxiaoxuan273) [reply registry.go:82]: changed accordingly.

File: registry.go (hunk 2 above)
Comment (wangxiaoxuan273) [reply registry.go:92]: changed accordingly.

File: registry.go [hunk 3]
```diff
@@ -71,28 +71,30 @@ func Logout(ctx context.Context, store Store, registryName string) error {
...
 }
 
-func mapStoreRegistryName(registry string) string {
-	// The Docker CLI expects that the 'docker.io' credential
-	// will be added under the key "https://index.docker.io/v1/"
-	// See: https://github.com/moby/moby/blob/v24.0.0-beta.2/registry/config.go#L25-L48
+// ServerAddressFromRegistry maps a registry to a server address, which is used as
+// a key for credentials store. The Docker CLI expects that the credentials of
+// the registry 'docker.io' will be added under the key "https://index.docker.io/v1/".
+// See: https://github.com/moby/moby/blob/v24.0.0-beta.2/registry/config.go#L25-L48
```
[the remark this answers is not in the document]
Comment (wangxiaoxuan273) [reply registry.go]: updated the version number.

**Pull request merged**